### PR TITLE
fix(React 16): fix prerendering (bug 1435920)

### PR DESCRIPTION
**Pull request by dmose**
To test: 

1. `npm run cleanmc && npm run buildmc`
2. note that line 12 of `system-addon/prerendered/locales/en-US/activity-stream-prerendered.html` (the line with `div id="root"`) once again has lots of stuff in it.

### CODE REVIEW

File: system-addon/content-src/activity-stream-prerender.jsx [hunk 1]
```diff
@@ -19,16 +19,26 @@ export function prerenderStore() {
...
       <Base
         isPrerendered={true}
         locale={locale}
         strings={strings} />
-    </Provider>),
+    </Provider>);
+
+  // If this happens, it means pre-rendering is effectively disabled, so we
+  // need to sound the alarms:
+  if (!html || !html.length) {
```
Comment (k88hudson) [inline system-addon/content-src/activity-stream-prerender.jsx]: Good call!

File: system-addon/content-src/components/Base/Base.jsx [hunk 2]
```diff
@@ -77,10 +77,6 @@ export class _BaseContent extends React.PureComponent {
 
     const outerClassName = `outer-wrapper${shouldBeFixedToTop ? " fixed-to-top" : ""}`;
 
-    if (!props.isPrerendered && !initialized) {
```
Comment (k88hudson) [inline system-addon/content-src/components/Base/Base.jsx]: Can you explain this change?

File: system-addon/test/unit/activity-stream-prerender.test.jsx [hunk 3]
```diff
@@ -35,4 +35,16 @@ describe("prerender", () => {
...
+  it("should throw if zero-length HTML content is returned", () => {
+    const boundPrerender = prerender.bind(null, "en-US", messages, () => "");
+
+    assert.throws(boundPrerender, Error, /no HTML returned/);
+  });
+
+  it("should throw if falsy HTML content is returned", () => {
+    const boundPrerender = prerender.bind(null, "en-US", messages, () => null);
+
+    assert.throws(boundPrerender, Error, /no HTML returned/);
```
Comment (k88hudson) [inline system-addon/test/unit/activity-stream-prerender.test.jsx]: Thank you!

File: system-addon/test/unit/content-src/components/Base.test.jsx [hunk 4]
```diff
@@ -12,6 +12,12 @@ describe("<Base>", () => {
     assert.ok(wrapper.exists());
   });
 
+  it("should render the BaseContent component, passing through all props", () => {
+    const wrapper = shallow(<Base {...DEFAULT_PROPS} />);
+
+    assert.deepEqual(wrapper.find(BaseContent).props(), DEFAULT_PROPS);
+  });
+
```
Comment (k88hudson) [inline system-addon/test/unit/content-src/components/Base.test.jsx]: 👍

File: system-addon/content-src/components/Base/Base.jsx (hunk 2 above)
Comment (dmose) [reply system-addon/content-src/components/Base/Base.jsx]: It's not necessary; there is already the exact same guard in the (parent) Base component, so it doesn't need to be in BaseContent also.

**Comment by as-pine-proxy** [comment]
[Treeherder: bc043d5d19d0173aec2da7e9396e3cd2a613d096](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=bc043d5d19d0173aec2da7e9396e3cd2a613d096)

**Pull request merged**